File: CS50 Week 2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int cindex(string text);

int main(void)
{
    string input_text = get_string("Text: ");
    int index = cindex(input_text);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int cindex(string text)
{
    int letters = 0;
    int words = 1;
    int sentences = 0;
    int clindex;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        else if (text[i] == ' ')
        {
            words++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    clindex = round(0.0588 * letters / words * 100 - 0.296 * sentences / words * 100 - 15.8);
    return clindex;
}

File: CS50 Week 6/sentimental-readability/readability.py
import cs50

text = cs50.get_string("Text: ")
letters = 0
words = 1
sentences = 0
for i in range(len(text)):
    if (text[i].isalpha()):
        letters += 1
    elif (text[i] == " "):
        words += 1
    elif (text[i] in [".", "!", "?"]):
        sentences += 1
index = round(0.0588 * letters / words * 100 - 0.296 * sentences / words * 100 - 15.8)
if (index < 1):
    print("Before Grade 1")
elif (index > 16):
    print("Grade 16+")
else:
    print(f"Grade {index}")
